src/coreclr/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.CoreCLR.cs

```diff
@@ -188,17 +188,13 @@ public static int OffsetToStringData
             if (type is not RuntimeType rt)
             {
                 ArgumentNullException.ThrowIfNull(type);
-                throw new SerializationException(SR.Format(SR.Serialization_InvalidType, type));
+                static void Throw(Type type) => throw new SerializationException(SR.Format(SR.Serialization_InvalidType, type));
```

Comment:
What is this outlining trying to help with?

Reply:
Was just attempting to make the method more inlineable.

Reply:
It may not be needed or helpful, not sure.

Reply:
I do not see it helping with inling.

Test: 

```
[MethodImpl(MethodImplOptions.NoInlining)]
static void M(Type t) => RuntimeHelpers.GetUninitializedObject(t);
```

Disassembly:
```
; Assembly listing for method Program:<<Main>$>g__M|0_0(System.Type) (Tier1)
; Emitting BLENDED_CODE for X64 with AVX - Windows
; Tier1 code
; optimized code
; rsp based frame
; partially interruptible
; No PGO data

G_M000_IG01:                ;; offset=0x0000
       sub      rsp, 40
 
G_M000_IG02:                ;; offset=0x0004
       call     [System.Runtime.CompilerServices.RuntimeHelpers:GetUninitializedObject(System.Type):System.Object]
       nop      
 
G_M000_IG03:                ;; offset=0x000B
       add      rsp, 40
       ret      
 ```

If you would like to play inlining tricks, I think it would be better to do more in this method and less in the helpers so that all the fast path helpers are more likely to get inlined into this method. For example, change the last line to:

```
return rt.GetCreateUninitializedCache().CreateUninitializedObject(rt);
```

Reply:
Up to you really. I can make that change if you like. I'm fairly indifferent at this point.

Reply:
AggressiveInlining might help.

Reply:
I would just revert this change.

Reply:
Done